test/lib/test-helper.ts

```diff
@@ -1,64 +1,104 @@
+        serverOptions.push('--session-expiration=' + SESSION_EXPIRATION);
+        serverOptions.push('--longpoll-maxbuffersize=' + MAXBUFFERSIZE);
+        serverOptions.push('--longpoll-polltimeout=' + POLL_TIMEOUT);
+    }
 
-    server.stdout.on('data', (data: any): void => {
-        log(data.toString());
-    });
-    server.stderr.on('data', (data: any): void => {
-        log(data.toString());
-    });
-    server.on('close', (): void => {
-        log('child process exited');
-    });
+    var p: Promise<any> = https ? createCertificate() : Promise.resolve();
+    return p.then((): Promise<IOurSocket> => {
+        var server = child.spawn('node',
+                                 serverOptions,
+                                 {
+                                     stdio: [null, null, null, 'ipc']
+                                 }
+                                );
+        cleanupFns.push(((s: child.ChildProcess): void => {
+            s.kill();
+        }).bind(this, server));
+
+        server.stdout.on('data', (data: any): void => {
+            log(data.toString());
+        });
+        server.stderr.on('data', (data: any): void => {
+            log(data.toString());
+        });
+        server.on('close', (): void => {
+            log('child process exited');
+        });
 
-    return (new Promise<ReturnValue>((resolve: (result: ReturnValue) => void,
-                                      reject: (error: Error) => void): void => {
-        server.on('message', (m: any): void => {
-            // Connect the ipc socket once test server is ready
-            var socket: any = io.connect('http://localhost:' + testport);
-            socket.emitAcknowledge = (event: string, arg?: any): Promise<void> => {
-                return (new Promise<void>((resolve: () => void,
-                                           reject: (error: Error) => void): void => {
-                    var argument = (2 === arguments.length) ? arg : null;
-                    socket.emit(event, argument, (): void => {
-                        resolve();
-                    });
-                }));
-            };
-            socket.on('connect', (): void => {
-                resolve({
-                    'server': server,
-                    'ipc': socket
+        return (new Promise<IOurSocket>((resolve: (result: IOurSocket) => void,
+                                         reject: (error: Error) => void): void => {
+            server.on('message', (m: any): void => {
+                // Connect the ipc socket once test server is ready
+                var socket: any = io.connect('http://localhost:' + testport);
+                socket.emitAcknowledge = (event: string, arg?: any): Promise<void> => {
+                    return (new Promise<void>((resolve: () => void,
+                                               reject: (error: Error) => void): void => {
+                        var argument = (2 === arguments.length) ? arg : null;
```

Comment:
I believe that this would be equivalent and independent from the position of 'arg' in the parameter list: 
var argument = arg || null;